Make icons actually appear in the UI

Crown0815 opened the pull request:
Even though the icons are part of the training, they do not appear in the final application with the code given here. By adding the file extension, the icons are now displayed as expected.

Crown0815 committed:
Make icons actually appear in the UI

Even though the icons are part of the training, they do not appear in the final application with the code given here.
By adding the file extension, the icons are now displayed as expected.

adegeo commented:
@Crown0815 What OS are you testing on where you don't see it working? And other information too, what version of VS are you using? How are the resources added to your project? what version of .NET MAUI are you using?

We've tested on mac directly, deployed from windows to mac, windows, android, etc and they're appearing fine without the .png.

Crown0815 commented:
@adegeo I am using Windows 11 Home Version 21H2. I am not using Visual Studio but JetBrains Rider. The resources are in the `Resources/Images` folder, as specified in the instructions. That folder appears to be included with the template I used, i.e., in the project file there is the instruction

```
<MauiImage Include="Resources\Images\*" />
```

I assume, that adds the Images as MauiImages to the project?

I am quite surprised that the icons appear for you, since they are even absent from all the screenshots shown in the instructions, such as this one

![](https://learn.microsoft.com/en-us/dotnet/maui/tutorials/notes-app/media/shell/final.png?view=net-maui-7.0)

and this one

![](https://learn.microsoft.com/en-us/dotnet/maui/tutorials/notes-app/media/note/final.png?view=net-maui-7.0)

Am I missing something?

After I changed the reference to `icon_about.png` and `icon_notes` I see actual icons in the tabs. I assumed that's where they were supposed to go?

![grafik](https://user-images.githubusercontent.com/36126706/211197956-b2a69ddb-4b24-42ac-b70e-eda3a88a2499.png)
(I changed my color scheme to gray instead of purple, but the icons are now there, in contrast to the screenshots from the tutorial).

adegeo commented:
Are you saying, too, that if you changed it back to `icon_notes` (no .png) but then changed the color scheme to gray, you see them?

adegeo commented:
I talked with the product team and they said the preferred way to reference images is to use the extension. So I'm going to go through the tutorials and update them. I'm going to close this PR simply because there are a lot more places to update the name of the file, there are the other parts of the tutorial, references in the articles, the samples you can download, and the part 2 tutorial series which is in progress.

Crown0815 commented:
> Are you saying, too, that if you changed it back to `icon_notes` (no .png) but then changed the color scheme to gray, you see them?

No the color scheme had no effect on the icons. That was an independent comment to just explain why the text in my example is gray and not purple

Crown0815 commented:
> I talked with the product team and they said the preferred way to reference images is to use the extension. So I'm going to go through the tutorials and update them. I'm going to close this PR simply because there are a lot more places to update the name of the file, there are the other parts of the tutorial, references in the articles, the samples you can download, and the part 2 tutorial series which is in progress.

That is fine with me. I am just glad this issue will be resolved.

@adegeo If you don't mind, I would like to follow the progress on the changes made. When the PR is created could you notify me so I can follow it, please.
